source/common/network/io_socket_error_impl.h

```diff
@@ -21,6 +24,9 @@ class IoSocketError : public Api::IoError {
   // deleter deleteIoError() below to avoid deallocating memory for this error.
   static IoSocketError* getIoSocketEagainInstance();
 
+  // Similar case for `IoErrorCode::NoSupport`
+  static IoSocketError* getIoSocketInvalidAddressInstance();
```

Comment:
Could you make this method private?

I have some worries about potential for misuse of this accessor; the only real use of this method is to access the singleton from IoSocketError::ioResultSocketInvalidAddress.

Reply:
fixed in 5d4e5a465e9e34df1ddbdbd8b9c55cd24e675e30